src/components/crossDevice/MobileFlow.js

```diff
@@ -33,15 +36,22 @@ class MobileFlow extends Component {
 
   onClientSuccess = () => {
     this.props.actions.setClientSuccess(true)
+    this.props.actions.mobileConnected(false)
   }
 
   onDisconnectPing = (data) => {
     this.sendMessage('disconnect pong', data.roomId)
+    this.props.actions.mobileConnected(false)
   }
 
-  render = (props) =>
-    this.props.clientSuccess ?
-      <CrossDeviceSubmit {...props}/> : <MobileConnected {...props}/>
+  render = (props) => {
+    if (this.props.clientSuccess)
+      return <CrossDeviceSubmit {...props}/>
+    if (this.props.mobileConnected)
+      return <MobileConnected {...props}/>
+
+    return <MobileNotificationSent {...props}/>
```

Comment:
Minor: This is a slightly dangerous default return value. Can we refactor so we don't have a default and we error if we are in an unexpected state?